sound/soc/sof/pm.c

```diff
@@ -260,15 +260,61 @@ static void sof_cache_debugfs(struct snd_sof_dev *sdev)
 }
 #endif
 
+/*
+ * DSP D0/D3 States may transform as below:-
+ *
+ *       -------------------
+ *       |            d0i0 |
+ *       |       D0        |<--------------+
+ *       |            d0i3 |               |
+ *       -------------------               |
+ *          ^          |                   |
+ *          |          |suspend            |
+ *          |          v                   |
+ *          |  -------------------         |
+ *   resume |  | kwd_at_s0i3 &&  |   Yes   |
+ *          |  | target_is_s0i3  |---------+
```

Comment:
No, please don't mix D and S states this way. The state machine starts with S states which then impacts D states, you are doing things in the wrong order/direction.